### PR TITLE
Update sample env.

### PR DESCRIPTION
**Pull request by b95505017**

### CODE REVIEW

**Comment by florina-muntenescu** [comment]
@b95505017 could you do this for all samples, not just for the GithubBrowser sample, please?

**Comment by b95505017** [comment]
@florina-muntenescu Done.

File: BasicSample/app/build.gradle [hunk 1]
```diff
@@ -48,29 +48,24 @@ android {
...
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    compile 'com.android.support:appcompat-v7:' + rootProject.supportLibVersion;
-    compile 'com.android.support:cardview-v7:' + rootProject.supportLibVersion;
-    compile 'com.android.support:recyclerview-v7:' + rootProject.supportLibVersion;
-    compile 'android.arch.lifecycle:extensions:' + rootProject.archLifecycleVersion;
-    compile 'android.arch.persistence.room:runtime:' + rootProject.archRoomVersion;
-    annotationProcessor "android.arch.lifecycle:compiler:" + rootProject.archLifecycleVersion;
-    annotationProcessor "android.arch.persistence.room:compiler:" + rootProject.archRoomVersion;
+    compile 'com.android.support:design:' + rootProject.supportLibVersion
```
Comment (florina-muntenescu) [inline BasicSample/app/build.gradle]: these are still using `compile` whereas in other project you changed it to `implementation`

File: PersistenceContentProviderSample/app/build.gradle [hunk 2]
```diff
@@ -47,17 +47,17 @@ dependencies {
 
     // Test
     testCompile 'junit:junit:4.12'
-    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+    androidTestCompile('com.android.support.test.espresso:espresso-core:3.0.1', {
```
Comment (florina-muntenescu) [inline PersistenceContentProviderSample/app/build.gradle]: change `compile` to `Implementation` everywhere

File: BasicSample/app/build.gradle (hunk 1 above)
Comment (b95505017) [reply BasicSample/app/build.gradle]: There are only two projects use new Android plugin: `BasicRxJavaSampleKotlin` & `ithubBrowserSample`. `implementation` should be used only with new Android plugin.
Other projects still use stable plugin, so they should stick with `compile`.

File: PersistenceContentProviderSample/app/build.gradle (hunk 2 above)
Comment (b95505017) [reply PersistenceContentProviderSample/app/build.gradle]: `PersistenceContentProviderSample` use stable Android plugin (2.3.3), so `implementation` is not available.
Should I change all projects to beta Android plugin (3.0.0-beta3)?

Comment (b95505017) [reply PersistenceContentProviderSample/app/build.gradle]: @florina-muntenescu Any plan to merge?

**Pull request merged**